src/sentry/profiles/task.py

```diff
@@ -106,13 +106,28 @@ def _symbolicate(profile: MutableMapping[str, Any], project: Project) -> Mutable
     while True:
         try:
             response = symbolicator.process_payload(stacktraces=stacktraces, modules=modules)
-            for original, symbolicated in zip(
-                profile["sampled_profile"]["samples"], response["stacktraces"]
+
+            # make sure we're getting the same number of stacktraces back
+            assert len(profile["sampled_profile"]["samples"]) == len(response["stacktraces"])
```

Comment:
All `asserts` here will cause the task to silently fail, let's add a `sentry_sdk.capture_exception` somewhere to reflect the failure.